man/notify-selfcontained-example.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: MIT-0
+#
+# Implement the systemd notify protocol without external dependencies.
+# Supports both readiness notification on startup and on reloading,
+# according to the protocol defined at:
+# https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html
+# This protocol is guaranteed to be stable as per:
+# https://systemd.io/PORTABILITY_AND_STABILITY/
+
+import errno
+import os
+import signal
+import socket
+import time
+
+reloading = False
+terminating = False
+
+
+def notify(message):
+    if message is None or len(message) == 0:
+        raise OSError(errno.EINVAL, "notify() requires a message")
+
+    # Do nothing if no socket is available.
+    if "NOTIFY_SOCKET" not in os.environ:
+        return
+
+    socket_path = os.environ["NOTIFY_SOCKET"]
+    if socket_path[0] != "/" and socket_path[0] != "@":
+        raise OSError(errno.EAFNOSUPPORT, "Unsupported socket type")
+
+    # Handle abstract socket.
+    if socket_path[0] == "@":
+        socket_path = "\0" + socket_path[1:]
+
+    with socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM | socket.SOCK_CLOEXEC) as sock:
+        sock.connect((socket_path))
+        sock.sendall(message)
+
+
+def notify_ready():
+    notify(b"READY=1")
+
```

Comment:
Please drop the doubled empty lines between functions. This will be embedded in a man page, and we want to limit the amount of lines reuquired (both horizontally and vertically).

Reply:
Double spacing has been removed, there's only one line that's longer than 80 characters (the `with socket.socket...` line) - that could be shortened by importing a few things from the socket namespace rather than just importing the module, if you'd like.